ResidentInformationApi/V1/UseCase/ListContactsUseCase.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using ResidentInformationApi.V1.Boundary.Requests;
+using ResidentInformationApi.V1.Boundary.Responses;
+using ResidentInformationApi.V1.Factories;
+using ResidentInformationApi.V1.Gateways;
+
+namespace ResidentInformationApi.V1.UseCase
+{
+    public class ListContactsUseCase : IListContactsUseCase
+    {
+        private readonly IAcademyInformationGateway _academyGateway;
+        private readonly IHousingInformationGateway _housingGateway;
+        private readonly IMosaicInformationGateway _mosaicGateway;
+
+        public ListContactsUseCase(
+            IAcademyInformationGateway academyGateway,
+            IHousingInformationGateway housingGateway,
+            IMosaicInformationGateway mosaicGateway
+            )
+        {
+            _academyGateway = academyGateway;
+            _housingGateway = housingGateway;
+            _mosaicGateway = mosaicGateway;
+        }
+
+        public async Task<ResidentInformationResponse> Execute(ResidentQueryParam rqp)
+        {
+            var academyClaimants = await _academyGateway.GetClaimantInformation(rqp).ConfigureAwait(true);
+            var housingResidents = await _housingGateway.GetResidentInformation(rqp).ConfigureAwait(true);
+            var mosaicResidents = await _mosaicGateway.GetResidentInformation(rqp).ConfigureAwait(true);
+
+            var academyUrl = new Uri(Environment.GetEnvironmentVariable("ACADEMYAPI_URL"));
+            var housingUrl = new Uri(Environment.GetEnvironmentVariable("HOUSINGAPI_URL"));
+            var mosaicUrl = new Uri(Environment.GetEnvironmentVariable("MOSAICAPI_URL"));
+
+            var academyResults = academyClaimants.Select(x =>
+                new ResidentInformationResult
+                {
+                    System = "Academy",
+                    SystemId = x.ClaimId.ToString(),
+                    SystemUrl = academyUrl,
```

Comment:
I thought that the SystemUrl was supposed to be the URL that would be used to access the specific data point for this record. i.e. should be something like `https://the-academy-base-url/claimants/claim-id/1234` I can't remember what the actual academy fetch endpoint URL looks like but hopefully that makes sense

Reply:
Currently, the gateway doesn't return anything for the specific data point for the record, which is likely something we need to update, but may not be immediately applicable for this ticket.

Reply:
The urls are made up from ID's which should included in the response - is this not the case?

Reply:
Good point! I've updated based on the SwaggerHub docs for each individual api